pandas/core/categorical.py

```diff
@@ -474,13 +490,16 @@ def from_codes(cls, codes, categories, ordered=False):
             categorical. If not given, the resulting categorical will be
             unordered.
         """
+        from pandas import Index
+
         try:
             codes = np.asarray(codes, np.int64)
         except:
             raise ValueError(
                 "codes need to be convertible to an arrays of integers")
 
-        categories = cls._validate_categories(categories)
+        # have to use the instance, not property
```

Comment:
shouldn't this me a method on the Dtype itself? (which does't the coercion via Index)?

Reply:
> shouldn't this me a method on the Dtype itself? 

You mean the codes -> categories? Something like

```python
class CategoricalDtype:
    def _categories_from_codes(self, codes: Array[Int]) -> Array[Any]:
        ...
```

I'll try that out. The current way is a bit odd.

Reply:
I mean having to wrap an ``Index`` around it *here*; the ``_validate_categories`` should do that (IOW be able to accept a list-like), (and it should return an Index)

Reply:
maybe call

``CategoricalDtype._validate_categories(categories)`` I think its more clear